### PR TITLE
upgrade ES client

### PR DESCRIPTION
**Pull request by snopoke**
@esoergel

### CODE REVIEW

**Comment by snopoke** [comment]
maybe wait till docker build is merged rather than messing with travis setup to upgrade ES

**Comment by snopoke** [comment]
@dimagi/team-commcare-hq once this gets merged you'll be forced to upgrade your local ES to > 1.0.

It's a pretty simple process. Here are 3 options:

### Keep existing ES data
1. Stop old ES
2. Install ES 1.7.4
3. Update ES settings and point the data directory to your current ES data
4. Start new ES

OR

### Drop existing ES data and reindex
1. Stop old ES
2. Install and start new ES
3. run `./manage.py ptop_preindex --replace && ./manage.py ptop_es_manage --filp_all_aliases`

OR

### Drop existing data and reindex (using docker)
Same as above except don't install ES, use the docker version.
Use docker: [README](https://github.com/dimagi/commcare-hq/blob/master/docker/README.md)

File: corehq/ex-submodules/pillowtop/tests/test_elasticsearch.py [hunk 1]
```diff
@@ -156,17 +161,18 @@ def test_assume_alias(self):
...
 
     def test_assume_alias_deletes_old_aliases(self):
         # create a different index and set the alias for it
         pillow = TestElasticPillow()
-        new_index = 'test-index-with-duplicate-alias'
+        new_index = 'test_index-with-duplicate-alias'
         if not self.es.indices.exists(new_index):
             self.es.indices.create(index=new_index)
         self.es.indices.put_alias(new_index, pillow.es_alias)
+        self.addCleanup(functools.partial(ensure_index_deleted, new_index))
```
Comment (esoergel) [inline corehq/ex-submodules/pillowtop/tests/test_elasticsearch.py]: Nice, I hadn't seen `self.addCleanup` before.

File: settings.py [hunk 2]
```diff
@@ -743,7 +743,7 @@
 # on both a single instance or distributed setup this should assume localhost
 ELASTICSEARCH_HOST = 'localhost'
 ELASTICSEARCH_PORT = 9200
-ELASTICSEARCH_VERSION = 0.9
+ELASTICSEARCH_VERSION = 1.7
```
Comment (esoergel) [inline settings.py]: :exclamation:

**Comment by czue** [comment]
@snopoke out of curiosity do you recommend any one of those approaches more than others?

**Comment by snopoke** [comment]
I would encourage the 3rd option using docker but if you're not keen for that then the second

**Pull request merged**